src/shared/types/user.ts

```diff
@@ -1,4 +1,8 @@
 export interface User {
```

Comment:
declaring the type of something as `{ }` means it will always be just an empty object, which is not the case for progress and hist here.
edit this file to be the following:

```
export interface User {
  name: string
  email: string
  dateCreated: string
  type: string
  progress: Progress
  hist: Hist
}

interface Progress {
  // TODO
}

interface Hist {
  // TODO
}
```